sys-utils/ipcutils.c

```diff
@@ -227,6 +251,75 @@ void ipc_shm_free_info(struct shm_data *shmds)
 	}
 }
 
+int posix_ipc_shm_get_info(const char *name, struct posix_shm_data **shmds)
+{
+	DIR *d;
+	int i = 0;
+	struct posix_shm_data *p;
+	struct dirent *de;
+
+	p = *shmds = xcalloc(1, sizeof(struct posix_shm_data));
+	p->next = NULL;
+
+	d = opendir(_PATH_DEV_SHM);
+	if (!d)
+		err(EXIT_FAILURE, _("cannot open %s"), _PATH_DEV_SHM);
+
+	while ((de = readdir(d)) != NULL) {
+		if ((de->d_name[0] == '.' && de->d_name[1] == '\0') ||
+		    (de->d_name[0] == '.' && de->d_name[1] == '.' &&
+		     de->d_name[2] == '\0'))
+			continue;
+
+		if (strncmp(de->d_name, "sem.", 4) == 0)
+			continue;
+
+		struct stat st;
+		char path[PATH_MAX];
+		snprintf(path, sizeof(path), "%s/%s", _PATH_DEV_SHM, de->d_name);
+		if (stat(path, &st) < 0)
+			continue;
+		
+		memset(path, 0, sizeof(path));
+		path[0] = '/';
+		strcat(path, de->d_name);
+
```

Comment:
I see 
```
$ lsipc --posix-shmem
NAME                       PERMS OWNER SIZE MTIME
/lttng-ust-wait-8      rw-rw-rw-  kzak   4K Feb04
/lttng-ust-wait-8-1000 rw-r-----  kzak   4K Feb04
```

I do not think '/' is wanted, right? Would be enough `p->name = xstrdup(de->d_name);`.